test/test_operations.py

```diff
@@ -638,6 +638,28 @@ def test_pred_type(self):
     xla_f = xla_e.sum().item()
     self.assertEqual(f, xla_f)
 
+    # PRED cannot be automatically promoted to other dtypes.
+    # Test both cpu and xla here to make sure we get notified
+    # once PyTorch changes its behavior.
+    self.assertRaises(RuntimeError, lambda: c & c.byte())
+    self.assertRaises(RuntimeError, lambda: c + c.byte())
+    self.assertRaises(RuntimeError, lambda: xla_c & xla_c.byte())
+    # RuntimeError is trigger in execution instead of lowering,
+    # thus print is required.
+    self.assertRaises(RuntimeError, lambda: print(xla_c + xla_c.byte()))
```

Comment:
cc: @dlibenzi 
Testing assertRaises is tricky in lazy tensor approach as it's possible to be raise in either lowering process or graph execution process.